.github/workflows/pr-checks.yml

```diff
@@ -0,0 +1,34 @@
+name: Pull request checks
+
+on: [pull_request]
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Check formatting
+        run: cargo fmt --check
+
+  clippy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Run clippy
+        run: cargo clippy -- -Dwarnings
+
+  run-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Run unit tests
+        run: cargo test
```

Comment:
Do we want to break this job out into a separate workflow that runs for any push?

Reply:
Oh yeah that'd make sense! I'll do that.